src/workerd/api/streams/compression.c++

```diff
@@ -335,12 +390,11 @@ private:
       }
 
       // The pending read is still viable so determine how much we can copy in.
-      auto amountToCopy = kj::min(pending.buffer.size() - pending.filled, remaining);
-      std::copy(source, source + amountToCopy, pending.buffer.begin() + pending.filled);
-      source += amountToCopy;
+      auto amountToCopy = kj::min(pending.buffer.size() - pending.filled, output.size());
+      kj::ArrayPtr chunk = output.take(amountToCopy);
```

Comment:
nit ... I know the compiler will till in the type for the `kj::ArrayPtr chunk` here but I'd generally prefer this to either be explicit `kj::ArrayPtr<byte>` or `auto`. Non-blocking tho